### DISCUSSION TITLE
Am I able to build a GUI app which can display lines for a 3D modelling software?

### DISCUSSION THREAD

**Discussion by Joe23232** [fltk-rs/fltk-rs#602] [Q&A]
You see these lines in a 3D modelling software:

![image](https://user-images.githubusercontent.com/34926497/114023921-cadec900-98b6-11eb-9d8f-9cafad57c83a.png)

Am I able to make those kinds of lines using this software?

Additionally am I able to render a scene from something such as the [Bevy Game engine](https://bevyengine.org/) and then display it and then on the side I can add buttons using this `flkt-rs` crate?

For example:

![image](https://user-images.githubusercontent.com/34926497/114024448-6b34ed80-98b7-11eb-93c7-3a38f9757fc3.png)

---

I wanted to know am I able to build a UI similar to this:

![image](https://user-images.githubusercontent.com/34926497/114025234-41c89180-98b8-11eb-89f4-d9297b70d608.png)

**Comment by eadf** [comment]
You can display pretty much anything if you enable OpenGL in a part of your GUI, while still using the fltk widgets.
See examples here [opengl and more examples](https://github.com/MoAlyousef/fltk-rs-demos)

**Reply by Joe23232** [reply]
> You can display pretty much anything if you enable OpenGL in a part of your GUI, while still using the fltk widgets.
> See examples here [opengl and more examples](https://github.com/MoAlyousef/fltk-rs-demos)

See what I am planning to do is I want to display the contents from Bevy into this kind of editor, that is possible, right?

**Reply by eadf** [reply]
Ok, you can always _display_ 2d and 3d (offline)data from other sources like Bevy, but if you want to _run_ bevy in the same process as your fltk gui it's a little more complicated, and I'm not familiar with any of those details. Sorry

**Comment by Joe23232** [comment]
all good thanks
